src/test/shell/bazel/external_integration_test.sh

```diff
@@ -897,7 +905,7 @@ http_archive(
 )
 EOF
   bazel build @repo//... &> $TEST_log && fail "Expected to fail"
-  expect_log "Invalid SHA256 checksum"
+  expect_log $LOG
```

Comment:
This works by accident&mdash;and matches more than desired: the unquoted `$LOG` expands to several arguments and `expect_log` helpfully only takes the first argument, ignoring the rest. So this test verifies that the word `The` occurs in the log, which is not very specific.

On a more general point, it is often a bad idea to test for a precise error message; rewording a human-readable message should not cause tests to fail. It might be more meaningful to verify that certain pieces of information are present in the message, like the expected and the found hash sum, and the URL where the fetch was attempted from.